pkg/kv/kvserver/queue.go

```diff
@@ -910,6 +883,38 @@ func (bq *baseQueue) processLoop(stopper *stop.Stopper) {
 	}
 }
 
+// processOne processes a replica if possible and releases the processSem when
+// the processing is complete.
+func (bq *baseQueue) processOneAsync(
+	ctx context.Context, repl replicaInQueue, stopper *stop.Stopper,
+) {
```

Comment:
I wonder if we want something in the function name that indicates we assume the semaphore is held so we can see that clearly at the call site.